src/cascadia/TerminalApp/ShortcutActionDispatch.cpp

```diff
@@ -19,6 +19,19 @@ using namespace winrt::TerminalApp;
 
 namespace winrt::TerminalApp::implementation
 {
+    std::wstring ExtractAction(const ActionAndArgs& actionAndArgs)
+    {
+        // Formatted as "User.{action}" or "User.{action}.{hash}"
+        std::wstring_view remaining{ actionAndArgs.GenerateID() };

```

Comment:
This is unsafe, because string-views never hold onto the memory. They're just a view into someone else's container after all. Simply assign the return value from `GenerateID()` to a separate variable, which will ensure that the `hstring` is kept alive.

Reply:
man, I swear I'll learn `string_views` 😅. I tested it too and I thought it worked fine.

Reply:
That's because `GenerateID` currently happens to cache the ID and so it stays alive. But I don't think that's a detail we should rely on (because in a few years everyone will have forgotten about that).